app/controllers/admin/api/objects_controller.rb

```diff
@@ -0,0 +1,95 @@
+# frozen_string_literal: true
+
+class Admin::Api::ObjectsController < Admin::Api::BaseController
+
+  class AllowedStatusObject
+    attr_reader :controller, :status_object
+
+    def initialize(controller, status_object)
+      @controller = controller
+      @status_object = status_object
+    end
+
+    def authorize
+      raise NotImplementedError
+    end
+
+    private
+
+    def raise_access_denied
+      raise CanCan::AccessDenied
+    end
+  end
+
+  class BuyerAccountObject < AllowedStatusObject
+    def authorize
```

Comment:
As this will raise an error...

```suggestion
    def authorize!
```